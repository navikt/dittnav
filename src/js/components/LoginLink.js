import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage as F } from 'react-intl';

class LoginLink extends Component {
  render() {
    const { title, description, url, linkText, linkClassName } = this.props;
    return (
      <div className="element hodefot">
        <h3 className="decorated"><F id={title} /></h3>
        <p><F id={description} /></p>
        <a href={url} className={linkClassName}><F id={linkText} /></a>
      </div>
    );
  }
}

LoginLink.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  linkText: PropTypes.string.isRequired,
  linkClassName: PropTypes.string.isRequired,
};

export default LoginLink;
