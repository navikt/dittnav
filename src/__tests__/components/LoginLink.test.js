import * as React from 'react';
const ReactTestRenderer = require('react-test-renderer');
import LoginLink from 'js/components/LoginLink'
import wrapIntl from 'js/IntlTestHelper';

test('LoginLink renders', () => {
  const testData = {
    id: 'brukernavnpassord',
    title: 'paalogging.alternativ.brukernavnpassord.tittel',
    description: 'paalogging.alternativ.brukernavnpassord.beskrivelse',
    url: 'url',
    linkText: 'paalogging.alternativ.brukernavnpassord.lenketekst',
    linkClassName: 'customStyle',
  };

  const component = ReactTestRenderer.create(wrapIntl(<LoginLink {...testData} />));
  expect(component.toJSON()).toMatchSnapshot();
});
