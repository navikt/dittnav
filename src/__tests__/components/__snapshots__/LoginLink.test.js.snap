// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginLink renders 1`] = `
<div
  className="element hodefot"
>
  <h3
    className="decorated"
  >
    <span>
      Brukernavn og passord fra NAV
    </span>
  </h3>
  <p>
    <span>
      Pålogging med brukernavn og passord gir deg bare tilgang til selvbetjening for arbeidssøkere.
    </span>
  </p>
  <a
    className="customStyle"
    href="url"
  >
    <span>
      Velg
    </span>
  </a>
</div>
`;
